include/openssl/engine.h

```diff
@@ -801,6 +801,7 @@ typedef int (*dynamic_bind_engine) (ENGINE *e, const char *id,
             CRYPTO_set_mem_functions(fns->mem_fns.malloc_fn, \
```

Comment:
Just in case, you didn't notice, the line above reads:
```C
 if (ENGINE_get_static_state() == fns->static_state) goto skip_cbs; \
```
so this change is only effective it the ENGINE_get_static_state is different, which is not the normal case.

Reply:
so **by default** is not the case...

Reply:
So this is on the case where the ENGINE and the application are using different instances of libcrypto?

Reply:
Correct.

Reply:
I've noted that in the question to OTC above as that is a significant mitigating factor.